framework/sessions/utils.py

```diff
@@ -11,7 +10,8 @@ def remove_sessions_for_user(user):
     """
     from osf.models import Session
 
-    Session.remove(Q('data.auth_user_id', 'eq', user._id))
+    if user._id:
```

Comment:
Right! So there are several tests that create a user and then [set the password before saving it](https://github.com/CenterForOpenScience/osf.io/blob/develop/osf_tests/test_user.py#L765), resulting in user without an _id. Setting a password calls `remove_sessions_for_user` which then uses this query. Since `Q` was OK with querying on `None` and Django is not, I added the check for the `._id` before querying for a session to remove instead of changing every test that does this (there are 11 that do this) thinking that if a user isn't actually saved yet, there are no sessions to remove for realz. I could instead change the tests tho so lemmie know!

Reply:
I was wondering this too. Thanks for the explanation. I think this is fine as is.